Update license banners

Comandeer opened the pull request:
<!--
🚨 If you want to submit a PR for a security vulnerability, please contact us directly
at https://ckeditor.com/contact/ instead. 🚨
-->
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?

All patches that change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [ ] Unit tests
- [ ] Manual tests

## Did you follow the CKEditor 4 code style guide?

Your code should follow the guidelines from the [CKEditor 4 code style guide](https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md) which helps keep the entire codebase consistent.

- [ ] PR is consistent with the code style guide

## What is the proposed changelog entry for this pull request?

skip

## What changes did you make?

The PR contains updates of license banners in all files.

Comandeer commented:
I've updated the license update script to ignore git ignored files and files inside git submodules. It's helpful when running the script in other repos, e.g. `ckeditor4-docs`.

I've also added a simple `console.log()` to know which files are touched.

sculpt0r reviewed (changes requested):
Overall looks good. However, I found a few places that should be updated. The date was updated, but due to links - they don't match to old company regexp:
![image](https://user-images.githubusercontent.com/6498567/150766024-7847328e-2b67-4aab-960f-eaedd59f8d08.png)

Due to ☝️  maybe we can just update them manually without updating the regexp? Since tif we are going to ever change the company name again - it will be a different regexp anyway.

Comandeer commented:
> maybe we can just update them manually without updating the regexp?

Never! I've updated the regex accordingly :stuck\_out\_tongue:I've also added `.json` files to files allowed to be updated.
